### PR TITLE
Fail Project import if repo couldn't be `clone`d

### PR DESCRIPTION
**Pull request by yannickperrenet**
## Description

Properly handle `git clone` (e.g. specifying a private repo URL) failures during Project import:

https://user-images.githubusercontent.com/26223174/197776080-2245b982-d7c3-474d-b7d1-1c42b590733d.mp4

### CODE REVIEW

File: services/orchest-webserver/app/scripts/background_tasks.py [hunk 1]
```diff
@@ -33,7 +33,9 @@ def git_clone_project(args) -> Tuple[int, str]:
 
         # This way we clone in a directory with the name of the repo if
         # the project_name is not provided.
-        git_command = f"git clone {args.url}"
+        # - GIT_TERMINAL_PROMPT makes the `git` command fail if it has
```
Comment (mausworks) [inline services/orchest-webserver/app/scripts/background_tasks.py:36]: TIL! :fire:

File: services/orchest-webserver/app/scripts/background_tasks.py [hunk 2]
```diff
@@ -33,7 +33,9 @@ def git_clone_project(args) -> Tuple[int, str]:
...
+        #   to prompt the user for input, e.g. for their username.
+        git_command = f"GIT_TERMINAL_PROMPT=0 git clone {args.url}"
```
Comment (yannickperrenet) [inline services/orchest-webserver/app/scripts/background_tasks.py:38]: Note that we just fail whenever a prompt is issued by `git`. This does not distinguish between cases as to why the `git clone` failed. Felt unnecessary for now, it is already a good improvement of the current implementation.

File: services/orchest-webserver/client/src/projects-view/ImportDialog.tsx [hunk 3]
```diff
@@ -53,6 +53,8 @@ import React from "react";
 import { useImportGitRepo, validProjectName } from "./hooks/useImportGitRepo";
 
 const ERROR_MAPPING: Record<CreateProjectError, string> = {
+  "git clone failed":
+    "failed to clone repo. Only public repos can be imported.",
```
Comment (mausworks) [inline services/orchest-webserver/client/src/projects-view/ImportDialog.tsx:57]: Nit: Not sure if intended but it looks weird when the F isn't capitalized in "Failed", and then when O is capitalized in "Only". :thinking:

**Review by mausworks — APPROVED** [review]
One small nit, otherwise LGTM. :+1:

File: services/orchest-webserver/client/src/projects-view/ImportDialog.tsx (hunk 3 above)
Comment (yannickperrenet) [reply services/orchest-webserver/client/src/projects-view/ImportDialog.tsx:57]: The text is actually displayed as: `Import failed: failed to clone repo. Orchest can only import public repos.`

And not capitalizing is in line with the other errors that we display.

**Pull request merged**